media/client/main/source/WebAudioPlayer.cpp

```diff
@@ -60,83 +67,163 @@ std::unique_ptr<IWebAudioPlayer> WebAudioPlayerFactory::createWebAudioPlayer(std
+                               const std::shared_ptr<IWebAudioPlayerIpcFactory> &webAudioPlayerIpcFactory,
+                               const std::shared_ptr<ISharedMemoryManagerFactory> &sharedMemoryManagerFactory)
     : m_webAudioPlayerClient(client)
 {
     RIALTO_CLIENT_LOG_DEBUG("entry:");
+    m_sharedMemoryManager = sharedMemoryManagerFactory->getSharedMemoryManager();
+    if (!m_sharedMemoryManager)
+    {
+        throw std::runtime_error("Shared memory manager could not be created");
+    }
+
+    if (!m_sharedMemoryManager->registerClient(this))
+    {
+        throw std::runtime_error("Failed to register client with SharedMemoryManager");
+    }
+
+    if (!webAudioPlayerIpcFactory)
+    {
+        throw std::runtime_error("Web audio player ipc factory could not be null");
+    }
+
+    m_webAudioPlayerIpc = webAudioPlayerIpcFactory->createWebAudioPlayerIpc(this, audioMimeType, priority, config);
+    if (!m_webAudioPlayerIpc)
+    {
+        (void)m_sharedMemoryManager->unregisterClient(this);
+        throw std::runtime_error("Web audio player ipc could not be created");
+    }
 }
 
 WebAudioPlayer::~WebAudioPlayer()
 {
     RIALTO_CLIENT_LOG_DEBUG("entry:");
+
+    m_webAudioPlayerIpc.reset();
+    if (!m_sharedMemoryManager->unregisterClient(this))
+    {
+        RIALTO_CLIENT_LOG_WARN("Failed to unregister client with SharedMemoryManager");
+    }
+
+    m_sharedMemoryManager.reset();
 }
 
 bool WebAudioPlayer::play()
 {
     RIALTO_CLIENT_LOG_DEBUG("entry:");
 
-    return false;
+    return m_webAudioPlayerIpc->play();
 }
 
 bool WebAudioPlayer::pause()
 {
     RIALTO_CLIENT_LOG_DEBUG("entry:");
 
-    return false;
+    return m_webAudioPlayerIpc->pause();
 }
 
 bool WebAudioPlayer::setEos()
 {
     RIALTO_CLIENT_LOG_DEBUG("entry:");
 
-    return false;
+    return m_webAudioPlayerIpc->setEos();
 }
 
-bool WebAudioPlayer::getBufferAvailable(uint32_t &availableFrames, std::shared_ptr<WebAudioShmInfo> &webAudioShmInfo)
+bool WebAudioPlayer::getBufferAvailable(uint32_t &availableFrames, std::shared_ptr<WebAudioShmInfo> &)
 {
     RIALTO_CLIENT_LOG_DEBUG("entry:");
-
-    return false;
+    if (!m_webAudioShmInfo)
+    {
+        m_webAudioShmInfo = std::make_shared<WebAudioShmInfo>();
+    }
+    std::lock_guard<std::mutex> bufLocker(m_bufLock);
+    return m_webAudioPlayerIpc->getBufferAvailable(availableFrames, m_webAudioShmInfo);
 }
 
 bool WebAudioPlayer::getBufferDelay(uint32_t &delayFrames)
 {
     RIALTO_CLIENT_LOG_DEBUG("entry:");
 
-    return false;
+    return m_webAudioPlayerIpc->getBufferDelay(delayFrames);
 }
 
 bool WebAudioPlayer::writeBuffer(const uint32_t numberOfFrames, void *data)
 {
     RIALTO_CLIENT_LOG_DEBUG("entry:");
 
-    return false;
+    if (!m_webAudioShmInfo)
+    {
+        RIALTO_CLIENT_LOG_ERROR("Web audio shared info is null!");
+        return false;
+    }
+
+    uint32_t dataLength = numberOfFrames * 4;
```

Comment:
Shouldn't this be:

uint32_t dataLength = numberOfFrames * webAudioConfig->sampleSize

Reply:
Not quite sure if that will work, sample size was 16 or 32 in the poc but we always do numberOfFrames  x4?

Might be a good idea to get deliver what we know works for now and add a new subtask in the userstory to investigate this (and any other improvements) once we have integrated?

Reply:
yes could be,  we don't have "sampleSize" member in WebAudioShmInfo